### ISSUE TITLE
In remoting calls, combining the $using: scope with namespace variable notation is quietly ignored

### ISSUE THREAD

**Issue by mklement0** [PowerShell/PowerShell#16019]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Possibly related:  #9204

Using namespace variable notation in combination with the `$using:` scope works as expected with _jobs_; e.g.:

```powershell
$env:FOO = 'bar'
# Outputs 'bar', as expected.
Start-Job { $using:env:FOO } | Receive-Job -Wait -AutoRemoveJob
```

By contrast, in the context of _remoting_ it fails quietly (the reference evaluates to `$null`):

```powershell
# Run from an elevated session on a machine with remoting enabled.
$env:FOO = 'bar'
Invoke-Command -ComputerName . { $using:env:FOO }
```

### Expected behavior

`'bar'` should print.

### Actual behavior

Nothing is output.

### Environment data

```powershell
PowerShell Core 7.2.0-preview.9
```

**Comment by dwtaber** [comment]
Huh, this is interesting.  When using the `-Session` parameter rather than `-ComputerName`, `Invoke-Command {$using:env:foo}` _does_ return the expected value.

**Comment by dwtaber** [comment]
I've identified the proximate cause of this issue, but not yet the root cause.  If you're using the `-ComputerName` parameter, because the remote runspace isn't opened yet, the PowerShell version on the remote computer is assumed to be PowerShell v2:

https://github.com/PowerShell/PowerShell/blob/dad7a34ee743c81dd96e742ed5516ac782388972/src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs#L1900-L1945

Because PowerShell v2 doesn't understand `$Using:` expressions, [`GetPowerShellForPSv2()`](https://github.com/PowerShell/PowerShell/blob/dad7a34ee743c81dd96e742ed5516ac782388972/src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs#L2167-L2229) is supposed to alter the script before sending to the remote computer.  I'd guess that whatever parsing happens in [`GetConvertedScript`](https://github.com/PowerShell/PowerShell/blob/dad7a34ee743c81dd96e742ed5516ac782388972/src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs#L2343-L2401) doesn't properly account for cases like `$using:env:foo`, but I haven't had a chance to dig in yet.

To test whether I was on the right track, I modified [`CreatePipeline`](https://github.com/PowerShell/PowerShell/blob/master/src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs#L1870-L1898), replacing the conditional expression on lines 1882–1884 with `GetPowerShellForPSv3OrLater(PSv5OrLater)` to ignore version checking and assume the remote computer would use PowerShell v5 or later.  Sure enough, `Invoke-Command` started returning values for `$using:env:foo`, `$env:variable:foo`, etc.  Obviously it's not acceptable for production to just do away with version checking, but it at least helps pin down the cause.  I'll keep digging.
